src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -650,49 +669,12 @@ protected final void afterInternal(boolean afterClass) throws Exception {
         } finally {
             if (!success) {
                 // if we failed here that means that something broke horribly so we should clear all clusters
-                afterTestRule.forceFailure();
+                // TODO: just let the exception happen, WTF is all this horseshit
+                // afterTestRule.forceFailure();
             }
         }
     }
 
-    @Override
-    protected final AfterTestRule.Task afterTestTask() {
-        return new AfterTestRule.Task() {
-            @Override
-            public void onTestFailed () {
-                //we can't clear clusters after failure when using suite scoped tests, as we would need to call again
```

Comment:
I think this is still used by the suite cluster. The general idea was that when a test fails in some cases some subsequent test would fail too unless we reset the cluster (test isolation issue). That said it was more important when we had the global cluster. The worst thing that can happen if we remove it now is that a test fails and all the other tests in the same suite fail too because of that. The code is quite cumbersome but that is another story, you can read more about why we went this path (the rule etc.) on the original PR: #9015 . I tend to be +1 on removing it.